turbopack/crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -394,6 +399,27 @@ impl DiskFileSystemInner {
 
         Ok(())
     }
+
+    async fn create_directory(self: &Arc<Self>, directory: &Path) -> Result<()> {
+        if !self.created_directories.contains(directory) {
+            let func = {
+                let inner = self.clone();
+                move |p: &Path| -> io::Result<()> {
+                    std::fs::create_dir_all(p)?;
+                    inner.created_directories.insert(p.to_path_buf());
+                    Ok(())
+                }
+            };
+            retry_blocking(directory, func)
+                .concurrency_limited(&self.semaphore)
+                .instrument(tracing::info_span!(
+                    "create directory",
+                    path = display(directory.display())
+                ))
+                .await?;
```

Comment:
I know you're just copying this, but we really shouldn't retry filesystem operations. They shouldn't fail transiently.

Reply:
How do you mean that?

At least on windows fs operations sometimes fail with permission errors while the antivir is scanning the file. That will automatically resolve by retrying.
We also do that in webpack using graceful-fs

Reply:
> At least on windows fs operations sometimes fail with permission errors while the antivir is scanning the file. That will automatically resolve by retrying.

WTF, that's insane. I've never seen a unix tool retry filesystem operations.

Is this maybe just for deletions that this can happen? I know Windows has a restriction that you can't unlink an open file. If that's the case, we should only retry deletions.

Reply:
v0 thinks retries makes sense on linux too

https://v0.dev/chat/KFWvUZ7IY9T